cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -619,9 +619,9 @@ updateOutputTemplate
     nonDelegCoin = fromIntegral totalSupply - delegCoin
 
     distribute :: Integer -> Int -> [AddressInEra ShelleyEra] -> [(AddressInEra ShelleyEra, Lovelace)]
-    distribute funds nAddrs addrs = zip addrs (fmap Lovelace (coinPerAddr + remainder:repeat coinPerAddr))
+    distribute funds nAddrs addrs = zip addrs $ Lovelace <$> (coinPerAddr + remainder:repeat coinPerAddr)
       where coinPerAddr, remainder :: Integer
-            (coinPerAddr, remainder) = funds `divMod` fromIntegral nAddrs
+            (coinPerAddr, remainder) = max 0 funds `divMod` fromIntegral nAddrs
```

Comment:
This is a stopgap. What we should do is have a check that errors if funds become negative.

Reply:
Added value check.